br/pkg/lightning/errormanager/errormanager.go

```diff
@@ -504,8 +534,143 @@ func (em *ErrorManager) ResolveAllConflictKeys(
+	}
+
+	g, gCtx := errgroup.WithContext(ctx)
+	pool.ApplyOnErrorGroup(g, func() error {
+		// check index KV first
+		rawHandleRows, err := em.db.QueryContext(
+			gCtx, fmt.Sprintf(selectIndexConflictKeysReplace, em.schemaEscaped),
+			tableName)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		for rawHandleRows.Next() {
+			var rawKey []byte
+			if err := rawHandleRows.Scan(&rawKey); err != nil {
+				return errors.Trace(err)
+			}
+			em.logger.Debug("got rawKey from table",
+				zap.ByteString("rawKey", rawKey))
+
+			var value []byte
+			value, err = fnGetLatest(gCtx, rawKey)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			var rows *sql.Rows
+			rows, err = em.db.QueryContext(
+				gCtx, fmt.Sprintf(selectConflictKeysReplaceByRawKey, em.schemaEscaped),
+				tableName, rawKey)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			for rows.Next() {
+				var indexName string
+				var rawValue []byte
+				if err := rows.Scan(&indexName, &rawValue); err != nil {
+					return errors.Trace(err)
+				}
+				em.logger.Debug("got index_name, raw_value from table",
+					zap.String("index_name", indexName),
+					zap.Binary("raw_value", rawValue),
+					zap.Binary("latest value", value))
+				if bytes.Equal(rawValue, value) {
+					continue
+				}
+				indexInfo := tbl.Meta().FindIndexByName(indexName)
+				var handle tidbkv.Handle
+				handle, err = decoder.DecodeHandleFromIndex(indexInfo, rawKey, rawValue)
+				em.logger.Debug("got handle from index",
+					zap.String("handle", handle.String()),
+					zap.Error(err))
+				rowKey := tablecodec.EncodeRowKeyWithHandle(tbl.Meta().ID, handle)
+				em.logger.Debug("got rowKey from handle",
+					zap.String("rowKey", rowKey.String()))
+				var overwrittenRow []byte
+				overwrittenRow, err = fnGetLatest(gCtx, rowKey)
+				if tikverr.IsErrNotFound(err) {
+					continue
+				}
+				if err != nil {
+					return errors.Trace(err)
+				}
+				var decodedData []types.Datum
+				decodedData, _, err = tables.DecodeRawRowData(encoder.SessionCtx, tbl.Meta(), handle, tbl.Cols(), overwrittenRow)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				_, err = encoder.Table.AddRecord(encoder.SessionCtx, decodedData)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				kvPairs := encoder.SessionCtx.TakeKvPairs()
+				for _, kvPair := range kvPairs.Pairs {
+					em.logger.Debug("got encoded KV",
+						logutil.Key("key", kvPair.Key),
+						zap.Binary("value", kvPair.Val),
+						logutil.Key("rawKey", rowKey),
+						zap.Binary("rawValue", rawValue))
+					if bytes.Equal(kvPair.Key, rawKey) && bytes.Equal(kvPair.Val, rawValue) {
+						if err := fnDeleteKey(gCtx, rowKey); err != nil {
+							return errors.Trace(err)
+						}
+						break
+					}
+				}
+			}
+			if err := rows.Err(); err != nil {
+				return errors.Trace(err)
+			}
+			if err := rows.Close(); err != nil {
+				return errors.Trace(err)
+			}
+		}
+		if err := rawHandleRows.Err(); err != nil {
```

Comment:
maybe we only need to call Close() 🤔

Reply:
if we delete `rows.Err()`, CI would report error:
```
br/pkg/lightning/errormanager/errormanager.go:566:40: rows.Err must be checked (rowserrcheck)
```
so I add back `rows.Err()`